test/integration/tags_test.exs

```diff
@@ -329,4 +329,38 @@ defmodule Solid.Integration.TagsTest do
              """
     end
   end
+
+  describe "custom tag" do
+    defmodule FoobarTag do
```

Comment:
One thing that I'm not sure is how this interface will handle custom block tags: 

https://github.com/Shopify/liquid/blob/e83b1e415990894c9517f94a8c2020ff825da027/test/unit/block_unit_test.rb#L61-L71

Not something we need to solve now but it will probably break this interface pre 1.0.0

Reply:
Good question. I was thinking about this and it _might_ need to work similarly to `capture`. We could add a signifier that a block is expected (like `{% custom do %} magic {% endcustom %}`) so it'd both know to expect a block _and_ an endtag.

Reply:
Yeah the block will be a bit more complex as not just arguments will be allowed but I think any valid liquid? Hopefully we will have a more solid (no pun intended) interface to expose to custom tags. I was hoping to have actual structs instead of keywords like NimbleParsec does.